src/storage/background_process.cppm

```diff
@@ -35,10 +37,18 @@ public:
 private:
     void Process();
 
+    void UpdateCheckpointState(TxnTimeStamp max_commit_ts, i64 wal_size);
+
+private:
     BlockingQueue<SharedPtr<BGTask>> task_queue_;
     Thread processor_thread_{};
 
-    WalManager* wal_manager_{};
+    WalManager *wal_manager_{};
+    Catalog *catalog_{};
```

Comment:
Since BGTaskProcessor has the catalog ptr, the processor initialization should be after the catalog initialization.
Storage::Init(), L83, need to be move to L87.

Reply:
catalog is initialized in wal_mgr_->ReplayWalFile(), which is before bg_processor initialization.

Reply:
After builtin_functions.Init() the catalog initialization is finished.

Reply:
OK